### ISSUE TITLE
Sporadic issues "I couldn't merge this branch: GitLab failed to rebase the branch saying: Rebase failed. Please rebase locally"

### ISSUE THREAD

**Issue by okainov** [smarkets/marge-bot#222]
0. marge-bot is used with --rebase-remotely parameter to use Gitlab API for rebase.
1. Assign some MR to marge-bot which does not require rebase at all
2. Wait for picking up the MR
3. Sometimes marge will complain "I couldn't merge this branch: GitLab failed to rebase the branch saying: Rebase failed. Please rebase locally". However there is even no need to rebase at all, I have green "Merge" button in UI!

Also I observed the same issue when rebasing is in progress or CI is in progress - marge does not understand that CI is in progress.

**Comment by aschmolck** [comment]
Could you try running marge with debug output, and paste what actual Gitlab API call returns back (and also what version of gitlab you are using)? Not fully sure if there's a problem with Gitlab's rebase API or marge's use of it from this description.

**Comment by okainov** [comment]
Not really, we wrote our own solution and use it not instead of marge-bot. This was quite reproducible
